app/Http/Requests/Sharing/ListSharingRequest.php

```diff
@@ -38,15 +38,7 @@ class ListSharingRequest extends BaseApiRequest implements HasBaseAlbum, HasOpti
 	 */
 	public function authorize(): bool
 	{
-		if (Gate::check(UserPolicy::IS_ADMIN)) {
-			return true;
-		}
-
-		if (!Gate::check(UserPolicy::CAN_UPLOAD, User::class)) {
-			return false;
-		}
-
-		if ($this->album !== null && Gate::check(AlbumPolicy::IS_OWNER, $this->album)) {
+		if (Gate::check(AlbumPolicy::CAN_SHARE_WITH_USERS, [AbstractAlbum::class, $this->album])) {
```

Comment:
That is slightly incorrect.
`Auth::id()` returns `null` if unauthenticated. As a result this would just return `null === null` and the authorization would pass. :(

Reply:
Yes and no.

First of all that case won't occur because it simply impossible to share an album with the `null` user, i.e. with the anonymous user.

Secondly, let's assume this was possible, i.e. sharing an album with `null` user, in that case it would also be perfectly fine if the anonymous user was allowed to query which albums have been shared with the anonymous user.

Reply:
I disagree. This is the authorization logic to _list_ the sharing the sharing rules for an album.
We do not need this to be accessible publicly.